### ISSUE TITLE
Sidekiq Batches - Pending job got stuck and never executed again

### ISSUE THREAD

**Issue by siva-zype** [sidekiq/sidekiq#6535]
Ruby version: 3.0.1
Rails version: 6.14
Sidekiq / Pro / Enterprise version(s): Sidekiq Pro 5.3.1

<img width="1403" alt="Screenshot 2024-12-03 at 11 43 12 PM" src="https://github.com/user-attachments/assets/b170bb35-566d-49ce-97ca-d99db37facaa">

**Comment by siva-zype** [comment]
Is there any way to re-enqueue the pending jobs?

**Comment by mperham** [comment]
Often this is fixed by super_fetch after some time. Make sure you have super_fetch enabled.

If you click the start time, you should see the batch details including the JID of the pending job. Unfortunately that's all Sidekiq knows about it. You'd need to requeue a job with the same JID and BID.

Also, please upgrade to the latest 6.x and Pro 5.x: `bundle up sidekiq-pro`.

**Comment by siva-zype** [comment]
@mperham Actually, I already enabled super_fetch
```
Sidekiq.configure_server do |config|
  config.redis = { :url => Rails.env.development? ? "redis://localhost:6379/6" : ENV["REDIS_URL"] }
  config.super_fetch!
 end
 `````

**Comment by raybaxter** [comment]
@siva-zype I've been able to recover jobs like this by restarting Sidekiq.

**Comment by siva-zype** [comment]
@mperham 
I got stuck on this issue. This is not related to deployment.
While executing batch child jobs, sometimes these jobs are removed from the running job and not enqueued back immediatly. Unable to see those jobs anywhere in the dashboard except batch pending count.
After some time(30-45 minutes), those skipped child jobs are retried to process.
Do we have any way to retry those child jobs immediately if it's skipped on the runtime.

**Comment by mperham** [comment]
No, there's no way to retry them immediately. The recovery process is not user-configurable. I'd suggest you track down those JIDs and determine what is unusual about them which causes them to disappear.
